playbooks/robusta_playbooks/oom_killer.py

```diff
@@ -1,30 +1,48 @@
+from datetime import timedelta
+
 from robusta.api import *
-from robusta.integrations.kubernetes.api_client_utils import parse_kubernetes_datetime
-from datetime import datetime, timezone
-from collections import namedtuple
+from robusta.integrations.resource_analysis.memory_analyzer import MemoryAnalyzer, K8sMemoryTransformer
 
 
-def is_last_state_in_oom_status(status: ContainerStatus):
-    if not status.lastState:
-        return False
-    if not status.lastState.terminated:
-        return False
-    return status.lastState.terminated.reason == "OOMKilled"
+class OomKillerEnricherParams(ActionParams):
+    """
+    :var new_oom_kills_duration_in_sec: In order to avoid duplicated, This playbook will only report OOMKills
+    that occurred in the last new_oom_kills_duration_in_sec seconds.
+    For example, if new_oom_kills_duration_in_sec is 1200, only OOMKills from the last 20 minutes will be considered.
+    """
+    new_oom_kills_duration_in_sec: int = 1200
 
+    """
+    :var prometheus_url: Prometheus url. If omitted, we will try to find a prometheus instance in the same cluster.
+    :example prometheus_url: "http://prometheus-k8s.monitoring.svc.cluster.local:9090".
+    """
+    prometheus_url: Optional[str] = None
 
-def is_state_in_oom_status(status: ContainerStatus):
-    if not status.state:
-        return False
-    if not status.state.terminated:
-        return False
-    return status.state.terminated.reason == "OOMKilled"
+    """
+    :var metrics_duration_in_secs: Memory usage metrics of nodes and pod containers where OOMKills have occurred will be queried
+    from prometheus in order to determine the estimated reason of each OOMKill.
+    This parameter determines the amount of time, in seconds, these metrics will be applied on.
+    For example, if duration_in_secs is 1200, memory usage metrics from the last 20 minutes will be considered.
+    """
+    metrics_duration_in_secs: int = 1200
 
+    """
+    :var container_memory_threshold: The maximal amount of memory percentage a pod container can use.
+    If this amount of memory was exceeded (in the last metrics_duration_in_secs seconds), the playbook will
+    report the corresponding container as the reason for the OOMKill.
+    """
+    container_memory_threshold: float = 0.92
 
-OOMKill = namedtuple("OOMKill", ["datetime", "message"])
+    """
+    :var node_memory_threshold: The maximal amount of memory percentage a node can use.
+    If this amount of memory was exceeded (in the last metrics_duration_in_secs seconds), the playbook will
+    report the corresponding node as the reason for the OOMKill.
+    """
+    node_memory_threshold: float = 0.95
 
 
 @action
-def oom_killer_enricher(event: NodeEvent):
+def oom_killer_enricher(event: NodeEvent, config: OomKillerEnricherParams):
```

Comment:
The event here should be a PrometheusEvent.
I don't think this enricher should be able to run on a NodeChange

Reply:
I agree, changed.

Also making sure the alert is of type HostOomKillDetected.

Reply:
I think that might be too much type validation. I think PrometheusAlert is good enough